### ISSUE TITLE
ORM m2m reverse bug

### ISSUE THREAD

**Issue by crackcomm** [astaxie/beego#671]
Hello.
I have tried to connect a model (`Movie`) with a model (`Person`) using two `Many-To-Many` relationships (`Directors` and `Writers`).

```Go
package models

import (
	"github.com/astaxie/beego/orm"
)

type Movie struct {
	Id          int    `json:"id"`
	Title       string `json:"title" orm:"size(200)"`
	
	// Many to many relationships
	Directors  []*Person   `orm:"rel(m2m);rel_through(github.com/crackcomm/myapp/models.MovieDirector)" json:"directors,omitempty"`
	Writers    []*Person   `orm:"rel(m2m);rel_through(github.com/crackcomm/myapp/models.MovieWriter)" json:"writers,omitempty"`
}

type MovieDirector struct {
	Id     int
	Movie  *Movie  `orm:"rel(fk)"`
	Person *Person `orm:"rel(fk)"`
}

type MovieWriter struct {
	Id     int
	Movie  *Movie  `orm:"rel(fk)"`
	Person *Person `orm:"rel(fk)"`
}

type Person struct {
	Id      int    `json:"id"`
	Name    string `orm:"size(200)" json:"name,omitempty"`
	
	Directed []*Movie `orm:"reverse(many);rel_through(github.com/crackcomm/myapp/models.MovieDirector)" json:"directed,omitempty"`
	Written  []*Movie `orm:"reverse(many);rel_through(github.com/crackcomm/myapp/models.MovieWriter)" json:"written,omitempty"`
}
```

Then when I'm trying to load related objects into a `Movie` model, I've got a `nil pointer dereference` error.

First of all - ORM maps `*Movie.Directors` to `*Person.Written` (it shouldn't).
And doesn't map `*Movie.Writers` reverse (this I think causes error).

```Go
movie := &models.Movie{}

o.QueryTable("movie").Filter("title", "Pulp Fiction").One(movie)

if err != nil {
	panic(err)
}

log.Printf("Movie: %v\n", movie)

o.LoadRelated(movie, "Directors")
o.LoadRelated(movie, "Writers")
```

After some code digging I came to conclusion that `*fieldInfo.reverseFieldInfoM2M` is a `nil` and probably causes an error.

```
D:/backup/go/src/github.com/astaxie/beego/orm/orm.go 345
D:/backup/go/src/github.com/astaxie/beego/orm/orm.go 297
D:/backup/go/src/github.com/astaxie/beego/orm/orm.go 208
```

PS.
All models are registered.
Before I tried `rel_through` i also tried `rel_table`.
This error appears on master branch (develop too, just checked).

**Comment by AyliD** [comment]
I am having similar problem... Any news on that?

**Comment by ADone** [comment]
I fixed this in that way #1250
It works in our project.

**Comment by crackcomm** [comment]
Fixed in #1381
